packages/sdk-middleware-http/src/http.js

```diff
@@ -146,23 +146,12 @@ export default function createHttpMiddleware({
                 return
               }
 
-              res.json().then((result: Object) => {
-                const parsedResponse: Object = {
-                  ...response,
-                  body: result,
```

Comment:
Can't we just do

```js
body: result.length > 0 ? JSON.parse(result) : {}
```

instead of extracting things into another function?

Reply:
That's much cleaner. I've updated the branch

Reply:
Test is also provided now